server/keycloak.ts

```diff
@@ -373,6 +402,60 @@ class Keycloak {
       throw error;
     }
   }
+
+  /**
+   * Migrate user's old keycloak id to new keycloak id and update roles
+   *
+   * @param keycloakId
+   * @param email
+   * @param username
+   */
+  async migrateUser(keycloakId: string, email: string, username: string): Promise<string[] | null> {
+    const migrationStatus = await dbClient.db[collections.USER_MIGRATION].findOne({
+      'email ilike': email,
+      'username ilike': username,
```

Comment:
A gotcha here; the username in the new realm may be `username@bceid_business` while on our database, just `username@bceid`.

Maybe specifically for BCeIDs, we should probably ignore the latter part of the username
We can't ignore it in all cases in case someone's IDIR `username2@idir` is the same as someone else's bceid `username2@bceid_business`